docs/topics/mpp/multiplatform-library.md

```diff
@@ -1,152 +1,161 @@
-object JvmBase64Encoder : Base64Encoder {
-    override fun encode(src: ByteArray): ByteArray = Base64.getEncoder().encode(src)
-}
+   > IDEA inspections help create `actual` implementations for an `expect` declaration.
+   >
+   {type="note"}
 
-```
+    ```kotlin
+    package org.jetbrains.base64
+    import java.util.*
+    
+    actual object Base64Factory {
+        actual fun createEncoder(): Base64Encoder = JvmBase64Encoder
+    }
+    
+    object JvmBase64Encoder : Base64Encoder {
+        override fun encode(src: ByteArray): ByteArray = Base64.getEncoder().encode(src)
+    }
+    ```
 
-Pretty simple, isn't it? We have provided a platform-specific implementation, but used a straightforward delegation to an implementation someone else has written!
+Pretty simple, right? You've provided a platform-specific implementation by using a straightforward delegation to a third-party implementation.
 
 ### JS
 
-Our JS implementation will be very similar to the JVM one. We create a file `Base64.kt` in `jsMain/kotlin/jetbrains/base64` and provide an implementation 
-which delegates to NodeJS `Buffer` API:
+The JS implementation will be very similar to the JVM one.
 
-```kotlin
-actual object Base64Factory {
-    actual fun createEncoder(): Base64Encoder = JsBase64Encoder
-}
+1. In the `jsMain/kotlin` directory, create the `org/jetbrains/base64` package.
+2. Create the `Base64.kt` file in the new package.
+3. Provide a simple implementation of the `Base64Factory` object that delegates to the NodeJS `Buffer` API:
 
-object JsBase64Encoder : Base64Encoder {
-    override fun encode(src: ByteArray): ByteArray {
-        val buffer = js("Buffer").from(src)
-        val string = buffer.toString("base64") as String
-        return ByteArray(string.length) { string[it].toByte() }
+    ```kotlin
+    package org.jetbrains.base64
+   
+    actual object Base64Factory {
+        actual fun createEncoder(): Base64Encoder = JsBase64Encoder
     }
-}
-```
+    
+    object JsBase64Encoder : Base64Encoder {
+        override fun encode(src: ByteArray): ByteArray {
+            val buffer = js("Buffer").from(src)
+            val string = buffer.toString("base64") as String
+            return ByteArray(string.length) { string[it].toByte() }
+        }
+    }
+    ```
 
 ### Native
 
-On the generic Native platform we don't have the luxury to use someone else's implementation, so we will have to write one ourselves. I won't explain the implementation details here,
-but it's pretty straightforward and follows Base64 format description without any optimizations:
-
-```kotlin
-private val BASE64_ALPHABET: String = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
-private val BASE64_MASK: Byte = 0x3f
-private val BASE64_PAD: Char = '='
-private val BASE64_INVERSE_ALPHABET = IntArray(256) {
-    BASE64_ALPHABET.indexOf(it.toChar())
-}
+Unfortunately, there is no third-party implementation available for Kotlin/Native, so you need to write it yourself.
```

Comment:
I suppose there _are_ implementations for Kotlin/Native targets, but no _generic_ one for all of them. Perhaps we can formulate this more precisely.

Reply:
Thanks. Rephrased.